go/cmd/dolt/commands/push.go

```diff
@@ -87,93 +89,128 @@ func (cmd PushCmd) Exec(ctx context.Context, commandStr string, args []string, d
+func constructInterpolatedDoltPushQuery(apr *argparser.ArgParseResults) (string, error) {
+	var params []interface{}
+	var args []string
+
+	if setUpstream := apr.Contains(cli.SetUpstreamFlag); setUpstream {
+		args = append(args, "'--set-upstream'")
 	}
-	var verr errhand.VerboseError
-	err = actions.DoPush(ctx, dEnv.RepoStateReader(), dEnv.RepoStateWriter(), dEnv.DoltDB, remoteDB, tmpDir, opts, buildProgStarter(defaultLanguage), stopProgFuncs)
-	if err != nil {
-		verr = printInfoForPushError(err, opts.Remote, opts.DestRef, opts.RemoteRef)
+	if force := apr.Contains(cli.ForceFlag); force {
+		args = append(args, "'--force'")
+	}
+	for _, arg := range apr.Args {
+		args = append(args, "?")
+		params = append(params, arg)
 	}
 
-	if opts.SetUpstream {
-		err := dEnv.RepoState.Save(dEnv.FS)
-		if err != nil {
-			err = fmt.Errorf("%w; %s", actions.ErrFailedToSaveRepoState, err.Error())
-		}
+	query := fmt.Sprintf("call dolt_push(%s)", strings.Join(args, ", "))
+	interpolatedQuery, err := dbr.InterpolateForDialect(query, params, dialect.MySQL)
+	if err != nil {
+		return "", err
 	}
 
-	return HandleVErrAndExitCode(verr, usage)
+	return interpolatedQuery, nil
 }
 
-func printInfoForPushError(err error, remote env.Remote, destRef, remoteRef ref.DoltRef) errhand.VerboseError {
-	switch err {
-	case doltdb.ErrUpToDate:
-		cli.Println("Everything up-to-date")
-	case doltdb.ErrIsAhead, actions.ErrCantFF, datas.ErrMergeNeeded:
-		cli.Printf("To %s\n", remote.Url)
-		cli.Printf("! [rejected]          %s -> %s (non-fast-forward)\n", destRef.String(), remoteRef.String())
-		cli.Printf("error: failed to push some refs to '%s'\n", remote.Url)
-		cli.Println("hint: Updates were rejected because the tip of your current branch is behind")
-		cli.Println("hint: its remote counterpart. Integrate the remote changes (e.g.")
-		cli.Println("hint: 'dolt pull ...') before pushing again.")
-		return errhand.BuildDError("").Build()
-	case actions.ErrUnknownPushErr:
-		status, ok := status.FromError(err)
-		if ok && status.Code() == codes.PermissionDenied {
-			cli.Println("hint: have you logged into DoltHub using 'dolt login'?")
-			cli.Println("hint: check that user.email in 'dolt config --list' has write perms to DoltHub repo")
-		}
-		if rpcErr, ok := err.(*remotestorage.RpcError); ok {
-			return errhand.BuildDError("error: push failed").AddCause(err).AddDetails(rpcErr.FullDetails()).Build()
-		} else {
-			return errhand.BuildDError("error: push failed").AddCause(err).Build()
+// getDefaultRemote gets the name of the default remote.
+func getDefaultRemote(sqlCtx *sql.Context, queryist cli.Queryist) (string, error) {
+	rows, err := GetRowsForSql(queryist, sqlCtx, "select name from dolt_remotes")
+	if err != nil {
+		return "", err
+	}
+	if len(rows) == 0 {
+		return "", env.ErrNoRemote
+	}
+	if len(rows[0]) == 0 {
+		return "", env.ErrNoRemote
+	}
+	if len(rows[0]) == 1 {
+		return rows[0][0].(string), nil
+	}
```

Comment:
if it's not 0 len, it will always be len == 1 because of you're query specifying the single column. Probably don't need to check for len == 1.